quickwit/quickwit-serve/src/lib.rs

```diff
@@ -604,7 +604,7 @@ pub async fn serve_quickwit(
     let (search_job_placer, search_service) = setup_searcher(
         &node_config,
         cluster.change_stream(),
-        metastore_through_control_plane.clone(),
+        metastore_client.clone(),
```

Comment:
this change is actually not _strictly_ necessary: even when using the proxied metastore client, the searchers don't get proxied because they only use metastore actions that go directly to the metastore (the proxied metastore client selectively decides which metastore queries to proxy or not)

Reply:
Sounds good. Can you update the doc of `ControlPlaneMetastore` and state which services use a proxied metastore and which don't.

Reply:
After a second thought, I actually reverted to using the `ControlPlaneMetastore` and just added a comment when providing ti to the the searcher initialization method. It doesn't make any difference today, but I prefer that the proxying rules are applied everywhere consistently, because an incomplete state in the control plane would be very hard to debug.